### PR TITLE
[EPMRPP-37422]  ItemAttribute repository, [EPMRPP-37420] IssueGroup repository, [EPMRPP-37419] IssueEntity repository, [EPMRPP-37416] Attribute repository

### PR DESCRIPTION
**Pull request by ihar-kahadouski**

### CODE REVIEW

File: src/test/java/com/epam/ta/reportportal/dao/suite/IssueEntityRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+	@FlywayTest(locationsForMigrate = { FILL_SCRIPT_PATH_ITEM }, invokeCleanDB = false)
+	@BeforeClass
+	public static void setUp() {
+	}
+
+	@Test
+	public void findAllByIssueId() {
+		final List<IssueEntity> issueEntities = repository.findAllByIssueTypeId(TO_INVESTIGATE_TYPE_ID);
+		assertEquals(EXPECTED_SIZE, issueEntities.size());
```
Comment (pbortnik) [inline src/test/java/com/epam/ta/reportportal/dao/suite/IssueEntityRepositoryTest.java]: I think that the rule of constant values is not applicable for tests. it is better to read test when assertion values are directly in the method. As an option, you can create a local variable in the method. Also, it is a good practice to add a message in assert.

Comment (pbortnik) [reply src/test/java/com/epam/ta/reportportal/dao/suite/IssueEntityRepositoryTest.java]: Also, if we search by a concrete issue type, is it a good idea to check that all issue enteties have the same issue type?

File: src/test/java/com/epam/ta/reportportal/dao/suite/IssueGroupRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+public class IssueGroupRepositoryTest extends BaseTest {
+
+	@Autowired
+	private IssueGroupRepository repository;
+
+	@Test
+	public void findByTestItemIssueGroup() {
+		Arrays.stream(TestItemIssueGroup.values()).filter(it -> !it.equals(TestItemIssueGroup.NOT_ISSUE_FLAG)).forEach(it -> {
+			final IssueGroup issueGroup = repository.findByTestItemIssueGroup(it);
+			assertEquals(it, issueGroup.getTestItemIssueGroup());
```
Comment (pbortnik) [inline src/test/java/com/epam/ta/reportportal/dao/suite/IssueGroupRepositoryTest.java]: Message

File: src/test/java/com/epam/ta/reportportal/dao/suite/IssueTypeRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+/**
+ * @author <a href="mailto:ihar_kahadouski@epam.com">Ihar Kahadouski</a>
+ */
+public class IssueTypeRepositoryTest extends BaseTest {
+
+	private static final String FILL_SCRIPT_PATH = "db/fill/issue-type";
+
+	private static final Long CUSTOM_ID = 100L;
+	private static final String CUSTOM_LOCATOR = "pb_ajf7d5d";
```
Comment (pbortnik) [inline src/test/java/com/epam/ta/reportportal/dao/suite/IssueTypeRepositoryTest.java]: move constants directly into test methods

File: src/test/java/com/epam/ta/reportportal/dao/suite/IssueTypeRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	@FlywayTest(locationsForMigrate = { FILL_SCRIPT_PATH }, invokeCleanDB = false)
+	@BeforeClass
+	public static void setUp() throws Exception {
+	}
+
+	@Test
+	public void findByLocator() {
+		final IssueType customIssueType = repository.findByLocator(CUSTOM_LOCATOR);
+		assertNotNull(customIssueType);
+		assertIssueType(customIssueType);
```
Comment (pbortnik) [inline src/test/java/com/epam/ta/reportportal/dao/suite/IssueTypeRepositoryTest.java]: give it more understendable name

File: src/test/java/com/epam/ta/reportportal/dao/suite/IssueTypeRepositoryTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+	@FlywayTest(locationsForMigrate = { FILL_SCRIPT_PATH }, invokeCleanDB = false)
+	@BeforeClass
+	public static void setUp() throws Exception {
+	}
+
+	@Test
+	public void findByLocator() {
+		final IssueType customIssueType = repository.findByLocator(CUSTOM_LOCATOR);
+		assertNotNull(customIssueType);
```
Comment (pbortnik) [inline src/test/java/com/epam/ta/reportportal/dao/suite/IssueTypeRepositoryTest.java]: add messages

File: src/test/java/com/epam/ta/reportportal/dao/suite/ItemAttributeRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+import static org.junit.Assert.*;
+
+/**
+ * @author <a href="mailto:ihar_kahadouski@epam.com">Ihar Kahadouski</a>
+ */
+public class ItemAttributeRepositoryTest extends BaseTest {
+
+	private static final String FILL_SCRIPT_PATH = "db/fill/item-attributes";
+
+	private static final String LAUNCH_KEY_NAME = "key1";
```
Comment (pbortnik) [inline src/test/java/com/epam/ta/reportportal/dao/suite/ItemAttributeRepositoryTest.java]: constants to methods

File: src/test/java/com/epam/ta/reportportal/dao/suite/ItemAttributeRepositoryTest.java [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+	@FlywayTest(locationsForMigrate = { FILL_SCRIPT_PATH }, invokeCleanDB = false)
+	@BeforeClass
+	public static void setUp() {
+	}
+
+	@Test
+	public void findByLaunchIdAndKeyAndSystem() {
+		final Optional<ItemAttribute> optionalAttr = repository.findByLaunchIdAndKeyAndSystem(LAUNCH_ID, LAUNCH_KEY_NAME, false);
+		assertTrue(optionalAttr.isPresent());
```
Comment (pbortnik) [inline src/test/java/com/epam/ta/reportportal/dao/suite/ItemAttributeRepositoryTest.java]: messages

File: src/test/resources/db/fill/issue-entity/V013__issue-entity-fill.sql [hunk 8]
```diff
@@ -0,0 +1,74 @@
+CREATE OR REPLACE FUNCTION items_init()
```
Comment (pbortnik) [inline src/test/resources/db/fill/issue-entity/V013__issue-entity-fill.sql]: let's write a little documentation in each fill script what kind of data it generates

File: src/test/java/com/epam/ta/reportportal/dao/suite/IssueGroupRepositoryTest.java (hunk 2 above)
Comment (ihar-kahadouski) [reply src/test/java/com/epam/ta/reportportal/dao/suite/IssueGroupRepositoryTest.java]: assertEquals generates message. Is it not enough?

**Pull request merged**